server/controllers/FollowController.js

```diff
@@ -0,0 +1,172 @@
+    if (followingId === followerId) {
+      return res.status(422).json({
+        errors: { message: 'You cannot follow yourself' },
+      });
+    }
+    // Get the user being followed
+    User.findById(followingId)
+      .then((user) => {
+      // Return error message if user is not found
+        if (!user) {
+          return res.status(404).json({
+            errors: { message: 'User not found' },
+          });
+        }
+        Follow.findOrCreate({ where: { followerId, followingId } })
+          .spread((userFollower, created) => {
+          // No action is performed if you already following an user
+            if (!created) {
+              return res.status(422).json({
+                errors: { message: 'You are already following this User' },
+              });
+            }
+            // Return user's profile
+            return res.status(201).json({
+              profile: profileHelper.userProfile(user, true),
+            });
+          });
+      }).catch(next);
+    return null;
+  }
+
+  /**
+   * remove a user from the follower list of an user
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static unfollow(req, res, next) {
+    const { userId: followingId } = req.params;
+    const { id: followerId } = req.body.decoded;
+
+    // Get the user being followed
+    User.findById(followingId)
+      .then((user) => {
+      // Return error message if user is not found
+        if (!user) {
+          return res.status(404).json({
+            errors: { message: 'User not found' },
+          });
+        }
+        // Remove user-follower relationship
+        user.removeFollower(followerId)
+          .then((follow) => {
+            if (!follow) {
+              return res.status(422).json({
+                errors: {
+                  message: 'You are not following this author',
+                },
+              });
+            }
+            // Return user's profile
+            return res.status(200).json({
+              profile: profileHelper.userProfile(user, false),
+            });
+          });
+      }).catch(next);
+    return null;
+  }
+
+  /**
+   * returns all the followers of a login user
+   * or the specified user
+   * @param {Object} req the request body
+   * @param {Object} res the response body
+   * @param {function} next a call to the next function
+   * @returns {Object} the response body
+   */
+  static followers(req, res, next) {
+    const otherUser = req.params.userId;
+    const userId = otherUser || req.body.decoded.id;
+
+    User.findById(userId)
+      .then((user) => {
+        if (!user) {
+          return res.status(404).json({
+            errors: { message: 'User not found' },
+          });
+        }
+        // get all the followers of the user
+        user.getFollowers()
```

Comment:
You can add the followers in an `include` while querying the `User` model. This should be more efficient than running 2 queries